### ISSUE TITLE
Doubt in kl divergence of iaf

### ISSUE THREAD

**Issue by npuichigo** [sbaurdlp/keras-iaf-mnist#1]
```python
# kl divergence
    # sampling for estimating the expectations
    for k in range(n_sample):
        epsilon = K.random_normal(shape=(batch_size, latent_dim), mean=0.,
                                  stddev=1.0)  # used for every z_i sampling
        z0_ = z_mean_0 + z_std_0 * epsilon
        z_ = z0_
        for z_mean, z_std in zip(z_means[1:], z_stds[1:]):
            z_ = iaf_transform_z([z_, z_mean, z_std])

        try:
            loss += K.sum(log_normal2(z0_, z_mean_0, 2 * K.log(z_std_0 + 1e-8)), -1)
        except NameError:
            loss = K.sum(log_normal2(z0_, z_mean_0, 2 * K.log(z_std_0 + 1e-8)), -1)
        loss -= K.sum(log_stdnormal(z_), -1)
```
First, the paper derives the posterior of latent variable given x ,
![image](https://user-images.githubusercontent.com/11533479/33709559-aa2825ac-db78-11e7-82b5-69c4dc47867a.png)
but how to derive the final kl divergence between the posterior and prior of z?
Second, in the last line `loss -= K.sum(log_stdnormal(z_), -1)`, the log likelihood of standard gaussian is negative, but why don't we use `loss += K.sum(log_stdnormal(z_), -1)` like above?
